flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -200,226 +125,224 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
-                        lines.push('---------------------------[color]white');
-
-                        const newMessage = { "id": Date.now(), "type": "FREE TEXT (" + sender + ")", "time": '00:00', "opened": null, "content": lines, };
-                        let timeValue = SimVar.GetGlobalVarValue("ZULU TIME", "seconds");
-                        if (timeValue) {
-                            const seconds = Number.parseInt(timeValue);
-                            const displayTime = Utils.SecondsToDisplayTime(seconds, true, true, false);
-                            timeValue = displayTime.toString();
-                        }
-                        newMessage["time"] = timeValue.substring(0, 5);
-                        this.messages.unshift(newMessage);
-                        toDelete.push(msg["id"]);
-                    }
+        CDUMenuPage.ShowPage(this.fmc, this);
+    }
 
-                    const msgCount = SimVar.GetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number");
-                    SimVar.SetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number", msgCount + toDelete.length).then();
-                })
-                .catch(err => {
-                    if (err.status === 404 || err === NXApi.disabledError || err === NXApi.disconnectedError) {
-                        return;
-                    }
-                    console.log("TELEX MSG FETCH FAILED");
-                });
-        }, NXApi.updateRate);
+    clearMessageQueue() {
+        this.messageQueue = [];
     }
 
     onUpdate(_deltaTime) {
-        super.onUpdate(_deltaTime);
-
-        if (this.pageUpdate) {
-            this.pageUpdate();
-        }
-        if (SimVar.GetSimVarValue("L:FMC_UPDATE_CURRENT_PAGE", "number") === 1) {
-            SimVar.SetSimVarValue("L:FMC_UPDATE_CURRENT_PAGE", "number", 0).then();
-            if (this.refreshPageCallback) {
-                this.refreshPageCallback();
+        if (this._delayedPageSwitch) {
+            this._delayedPageSwitch.timeRemaining -= _deltaTime;
+            if (this._delayedPageSwitch.timeRemaining < 0) {
+                this.updateRequest = false;
+                this._delayedPageSwitch.callback(this.fmc, this, this._delayedPageSwitch.args);
+                this._delayedPageSwitch = null;
             }
         }
 
-        this.checkAocTimes();
+        // the MCDUs are refreshed at a regular interval
+        if (this.pageRefreshThrottler.canUpdate(_deltaTime) !== -1) {
+            this.requestOnsideUpdate();
+        }
 
-        this.updateMCDU();
+        // and also on request (e.g. data changed), with a minimum time between updates
+        if (this.minPageUpdateThrottler.canUpdate(_deltaTime) !== -1 && this.updateRequest) {
+            this.updateRequest = false;
+            if (this.pageRedrawCallback) {
+                const delayedPageSwitch = this._delayedPageSwitch;
+                this.pageRedrawCallback();
+                this._delayedPageSwitch = delayedPageSwitch;
```

Comment:
**question**: It appears `this_delayedPageSwitch` might be modified by the `pageRedrawCallback`, but apparently this modification isn't allowed. It is however unclear to me as to why this is. Could you elaborate on this?

Reply:
I think it was a hack that was needed earlier on before some other parts were refactored... but probably not anymore... I will check.